### PR TITLE
Fix exception when using Get methods

### PR DESCRIPTION
**Pull request by willsb**
Fixes [this one](https://github.com/mono/mono/blob/master/mcs/class/System/System.Net/HttpWebRequest.cs#L845), to be more specific

Squash when: ever

### CODE REVIEW

File: Toggl.Ultrawave/Network/ApiClient.cs [hunk 1]
```diff
@@ -35,6 +33,17 @@ public void Dispose()
             httpClient.Dispose();
         }
 
+        private Task<HttpResponseMessage> sendAsync(IRequest request)
+        {
+            Ensure.ArgumentIsNotNull(request, nameof(request));
+
+            if (request.HttpMethod == HttpMethod.Get) 
+                return httpClient.GetAsync(request.Endpoint);
```
Comment (amulware) [inline Toggl.Ultrawave/Network/ApiClient.cs]: We only give it the endpoint? What about our own headers, and whatever other data the request holds?

**Comment by willsb** [comment]
X/Y problem ¯\_(ツ)_/¯

File: Toggl.Ultrawave/Network/ApiClient.cs [hunk 2]
```diff
@@ -41,9 +41,13 @@ private HttpRequestMessage createRequestMessage(IRequest request)
...
             var requestMessage = new HttpRequestMessage(request.HttpMethod, request.Endpoint);
             requestMessage.Headers.AddRange(request.Headers);
-            requestMessage.Content =
-                request.Body.Match<HttpContent>(json => new StringContent(json, Encoding.UTF8, "application/json"),
-                                                bytes => new ByteArrayContent(bytes));
+
+            if (request.HttpMethod != HttpMethod.Get)
+            {
+                requestMessage.Content =
+                    request.Body.Match<HttpContent>(json => new StringContent(json, Encoding.UTF8, "application/json"),
```
Comment (amulware) [inline Toggl.Ultrawave/Network/ApiClient.cs]: Do you mind splitting this differently? it's still quite long and feels very imbalanced. :D
I propose:
```
requestMessage.Content = request.Body.Match<HttpContent>(
    json => new StringContent(json, Encoding.UTF8, "application/json"),
    bytes => new ByteArrayContent(bytes)
);
```

Comment (willsb) [inline Toggl.Ultrawave/Network/ApiClient.cs]: Agreed

**Pull request merged**